src/ceph.py

```diff
@@ -796,3 +796,13 @@ def update(self):
         update_pool(client=self.service, pool=self.name, settings={"size": str(self.replicas)})
         # Perform other common post pool creation tasks
         super(ReplicatedPool, self).update()
+
+
+def get_osd_count():
+    """Return the number of OSDs."""
+    try:
+        ret = check_output(["ceph", "osd", "ls"])
+        return ret.decode("utf8").count("\n")
+    except Exception as e:
```

Comment:
I wonder if we should even handle this exception here. If this fails it's either a charm bug or the system is in a very bad state, and just erroring out might be best in both cases

Reply:
It was done just so that it can be called under any circumstances (i.e: cluster not bootstrapped, no monitors present, etc).

Reply:
Ok... maybe we could have a log message at WARNING level if we hit an exception here then?

Reply:
done.